### PR TITLE
Fix for failing secure check when 404 received

### PR DESCRIPTION
**Pull request by ps-jay**
This PR adds:
- a test for looping 301's (max redirects)
- a test for secure delivery of non-success HTTP codes
- a fix so that `be_verifiably_secure` passes if it receives a non-success HTTP code securely

### CODE REVIEW

File: lib/akamai_rspec/matchers/non_akamai.rb [hunk 1]
```diff
@@ -42,8 +42,15 @@ module Matchers
         rescue RestClient::MaxRedirectsReached
           return true # Securely sent a redirect
         rescue Exception => e
-          @error = e
-          return false
+          begin
+            if e.response # Securely sent a non-success HTTP code
+                          # See: https://github.com/rest-client/rest-client/tree/1.8.x#exceptions-see-wwww3orgprotocolsrfc2616rfc2616-sec10html
```
Comment (DanielHeath) [inline lib/akamai_rspec/matchers/non_akamai.rb]: I'm slightly concerned that this couples the HTTPS check with the RestClient internals.

I don't think there would ever be a 'truthy' response if the cert check failed though, so it *should* be alright.

Comment (ps-jay) [inline lib/akamai_rspec/matchers/non_akamai.rb]: Fair cop.

This is easier in rest-client v2.0 ...   but on v1.8, this approach is documented [here](https://github.com/rest-client/rest-client/tree/1.8.x#exceptions-see-wwww3orgprotocolsrfc2616rfc2616-sec10html) - so I think it should be okay.

Comment (ps-jay) [inline lib/akamai_rspec/matchers/non_akamai.rb]: I'm not sure how to mock an invalid cert in the rspec testing ...  could be interesting ...

Comment (DanielHeath) [inline lib/akamai_rspec/matchers/non_akamai.rb]: This could be rewritten as `e.respond_to?(:response) && e.response` instead of a begin/rescue/end block.

Comment (DanielHeath) [inline lib/akamai_rspec/matchers/non_akamai.rb]: Thinking about this further, I'm not sure `be_verifiably_secure` should use `rest-client` at all; ruby has quite reasonable https primitives built-in and rest-client assumes a little too much.

**Pull request merged**